examples/kelvin_helmholtz_instability.jl

```diff
@@ -0,0 +1,406 @@
+over the course of the `simulation`
+
+``
+energy(t₀ + Δτ) / energy(t₀) ≈ exp(2 σ Δτ)
+``
+
+where ``t₀`` is the starting time of the simulation and ``t₀ + Δτ`` 
+the ending time of the simulation. We thus find that the growth rate 
+is measured by
+
+``
+σ = log(energy(t₀ + Δτ) / energy(t₀)) / (2 * Δτ) .
+``
+"""
+function grow_instability!(simulation, energy)
+    ## Initialize
+    simulation.model.clock.iteration = 0
+    t₀ = simulation.model.clock.time = 0
+    compute!(energy)
+    energy₀ = energy[1, 1, 1]
+
+    ## Grow
+    run!(simulation)
+
+    ## Analyze
+    compute!(energy)
+    energy₁ = energy[1, 1, 1]
+    Δτ = simulation.model.clock.time - t₀
+
+    ## ½(u² + v²) ~ exp(2 σ Δτ)
+    σ = growth_rate = log(energy₁ / energy₀) / 2Δτ
+    return growth_rate    
+end
+nothing # hide
+
+# Finally, we write a function that rescales the state. The rescaling is done via measureing the
+# kinetic energy and then rescaling back to a prescribed energy value.
+# 
+# (Measuring the growth via the kinetic energy works fine _unless_ an unstable mode has _only_
+# buoynancy structure. In that case, the total perturbation energy will be adequate.)
+
+"""
+    rescale!(model, energy; target_kinetic_energy=1e-3)
+
+Rescales all model fields so that `energy = target_kinetic_energy`.
+"""
+function rescale!(model, energy; target_kinetic_energy=1e-6)
+    compute!(energy)
+
+    rescale_factor = √(target_kinetic_energy / energy[1, 1, 1])
+
+    model.velocities.u.data.parent .*= rescale_factor
+    model.velocities.v.data.parent .*= rescale_factor
+    model.velocities.w.data.parent .*= rescale_factor
+    model.tracers.b.data.parent .*= rescale_factor
+
+    return nothing
+end
+
+using Printf
+
+# Some more helper function for the power method,
+
+""" Compute the relative difference between ``σⁿ`` and ``σⁿ⁻¹``, avoiding `NaN`s. """
+relative_difference(σ) = length(σ) > 1 ? abs((σ[end] - σ[end-1]) / σ[end-1]) : Inf
+
+""" Check if the growth rate has converged. If the array `σ` has at least 2 elements then return the relative difference between ``σ[end]`` and ``σ[end-1]``. """
+convergence(σ) = length(σ) > 1 ? abs((σ[end] - σ[end-1]) / σ[end]) : 9.1e18 # pretty big (not Inf tho)
+nothing # hide
+
+# and the main function that performs the power method iteration.
+# (Note that `estimate_growth_rate!()` also return the plot of the perturbation field after each iteration.
+# This is not a requirement and can be disabled for speeding up the algorigithm but we'll keep it
+# here for illustration purposes.)
+"""
+    estimate_growth_rate!(simulation, energy, ω; convergence_criterion=1e-2)
+
+Estimates the growth rate iteratively until the relative change
+in the estimated growth rate ``σ`` falls below `convergence_criterion`.
+
+Returns ``σ``.
+"""
+function estimate_growth_rate!(simulation, energy, ω, b; convergence_criterion=1e-3)
+    σ = []
+    
+    plotseries = Array{Any, 1}(undef, 20)  # expect no more than 20 iterations
```

Comment:
I'm not happy with this but I couldn't think of anything else...
Perhaps `append()`?

Reply:
It's probably better to use 

```julia
movie_frames = []
push!(movie_frames, frame)
```

Note that `plotseries` and `powermethodplot` diverge from Oceananigans' style, which puts underscores `_` between words. In our style these are written `plot_series` and `power_method_plot`.

Reply:
I think your two options here are a) fetch the data and put it into an array or b) fetch the plot an put it into an array.

Usually I think we probably want to fetch data rather than embed plotting into the loop. To do that you might write

```julia
power_method_data = []

while convergence(σ) < convergence_criterion
    # ...
    push!(power_method_data, (ω=collect(interior(ω)[:, 1, :]), b=collect(interior(b)[:, 1, :]), σ=σ))
end
```

Later on, you can use this in plotting using `power_method_data[i].b`, etc (and you probably don't need a function to do the plotting; you can just write it out into the script.